src/ui/public/courier/__tests__/saved_object.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Tests functionality in ui/public/courier/saved_object/saved_object.js
+ */
+
+import ngMock from 'ng_mock';
+import expect from 'expect.js';
+import sinon from 'auto-release-sinon';
+
+import BluebirdPromise from 'bluebird';
+import SavedObjectFactory from '../saved_object/saved_object';
+import { stubMapper } from 'test_utils/stub_mapper';
+import IndexPatternFactory from 'ui/index_patterns/_index_pattern';
+
+describe('Saved Object', function () {
+  require('test_utils/no_digest_promises').activateForSuite();
+
+  let savedObjectFactory;
+  let IndexPattern;
+  let esStub;
+
+  beforeEach(ngMock.module('kibana'));
+  beforeEach(ngMock.inject(function ($injector, Private) {
+    savedObjectFactory = Private(SavedObjectFactory);
```

Comment:
I feel like I've more commonly seen this being imported as `SavedObject` and then used as a constructor in the `createSavedObject` below.

So instead of having:
```
let savedObject = {};
savedObjectFactory.call(savedObject, config);
return savedObject;
```

it'd become:
```
return new SavedObject(config);
```